frontend/src/lib/WorkflowParser.ts

```diff
@@ -205,13 +206,19 @@ export default class WorkflowParser {
     if (selectedWorkflowNode && selectedWorkflowNode.outputs) {
       (selectedWorkflowNode.outputs.artifacts || [])
         .filter((a) => a.name === 'mlpipeline-ui-metadata' && !!a.s3)
-        .forEach((a) =>
+        .forEach((a) => {
+          let source = StorageService.MINIO;
+          if (a.s3!.endpoint.indexOf('s3.amazonaws.com') >= 0){
+            source = StorageService.S3;
+          } else if (a.s3!.endpoint.indexOf('storage.googleapis.com') >= 0) {
```

Comment:
I'm surprised that this wasn't already here.
@yebrahim do you know why that was?

Reply:
Oh. I think this maybe unnecessary.. Here, it just returns minio artifacts and all cloud providers all use minio backed by PersistentVolume. So, all metadata artifacts are stored in minio. 

There're some discussion to use GCS or S3 to store metadata rather than using PV. This has not been finalized and we need more user requirements for it. Backend also need to make change. 

What I'd like to add is the support for s3 file we write into `mlpipeline-ui-metadata` artifact.

Reply:
Does it make sense? Let me remove this code snippet at this moment. It should be shipped with backend changes together once we determine to support different backends for metadata manifest

Reply:
Yup. IIRC, metadata is written inside the pod, and is extracted by Argo into minio, so there's no cloud provider specific code here.